pkg/clusteragent/externalmetrics/utils.go

```diff
@@ -96,3 +100,20 @@ func setQueryConfigValues(aggregator string, rollup int) {
 	queryConfigAggregator = aggregator
 	queryConfigRollup = rollup
 }
+
+func StructureIntoDDM(obj interface{}, structDest *v1alpha1.DatadogMetric) error {
```

Comment:
```suggestion
func UnstructuredIntoDDM(obj interface{}, structDest *v1alpha1.DatadogMetric) error {
```

Reply:
maybe  you can also take directly an `*unstructured.Unstructured` as parameter

Reply:
:+1: Done: https://github.com/DataDog/datadog-agent/pull/7857/commits/067fca31ab65d678c7b2056c1a3438ed22b16113